src/OpalCompiler-Core/ReservedVariable.class.st

```diff
@@ -48,6 +48,13 @@ ReservedVariable >> printOn: stream [
 	stream nextPutAll: self name
 ]
 
+{ #category : #queries }
+ReservedVariable >> usingMethods [
+	"first call is very slow as it creates all ASTs"
+	^SystemNavigation new allMethods select: [ : method |
+		method ast variableNodes anySatisfy: [ :varNode | varNode binding == self]]
```

Comment:
For those variables we need to implement optimized versions of readsIn:/writesIn: . Probably there is  already a way to ask method if it includes self sends, super calls or thisContext.
Then usingMethods will be quite trivial and fast.

But it is for another task. We need to clean and improve query protocol of variables.

Reply:
yes! There should be bytecode scanning method (or if not we can add them easily)